packages/reports/addon/components/filter-values/dimension-select.js

```diff
@@ -42,9 +50,14 @@ export default Ember.Component.extend({
    */
   dimensionOptions: computed('filter.subject', function() {
     let dimensionName = get(this, 'dimensionName'),
-      dimensionService = get(this, '_dimensionService');
+      dimensionService = get(this, '_dimensionService'),
+      metadataService = get(this, '_metadataService');
+
+    if (get(metadataService.getById('dimension', dimensionName), 'cardinality') <= LOAD_CARDINALITY) {
+      return dimensionService.all(dimensionName);
```

Comment:
this works fine for low cardinality dimensions since searching also waits for `all` to resolve, right?

Reply:
correct see: https://github.com/yahoo/navi/blob/master/packages/data/addon/services/bard-dimensions.js#L332